test/test_evaluator.cpp

```diff
@@ -558,4 +558,101 @@ TEST_F(ExprEvalTest, Permute) {
   EXPECT_THAT(out_tensor.strides(), ElementsAre(128, 1, 32, 8));
 }
 
+TEST_F(ExprEvalTest, ReshapePermuteReshape) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  TensorView* in =
+      TensorViewBuilder().shape({-1, 6}).dtype(DataType::Float).build();
+  fusion.addInput(in);
+  TensorView* out = reshape(
+      in, {size(in, 0), IrBuilder::create<Val>(2), IrBuilder::create<Val>(3)});
+  out = permute(out, {1, 2, 0});
+  out = reshape(out, {IrBuilder::create<Val>(6), size(out, 2)});
+  fusion.addOutput(out);
+
+  at::Tensor in_tensor = at::rand({72}).cuda().as_strided({9, 6}, {8, 1});
+
+  ExpressionEvaluator evaluator;
+  evaluator.bind(in, in_tensor);
```

Comment:
There is a bind operation in the `evaluate` method, so is this needed here?

Reply:
Good catch! In fact, it's the other way around. `ViewOp::evaluate` shouldn't call `bind`. [Code here](https://github.com/NVIDIA/Fuser/blob/6751f27a8e128c427f6fd4aedd58732e37fb4013/csrc/expr_evaluator.cpp#L201) finds dependencies and evaluates them recursively.

Reply:
Done.